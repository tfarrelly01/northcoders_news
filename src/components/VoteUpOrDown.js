import React from 'react';
import PropTypes from 'prop-types';

const VoteUpOrDown = props => (
  <figure>
    <p className="image is-96x96 has-text-centered">
      <img src={props.avatarUrl} alt="User Avatar" />
    </p>
    <section className="level-item has-text-centered ">
      <a 
        className="is-danger is-small row has-text-success" 
        onClick={props.handleVoteUp}
      >
        <i className="fa fa-thumbs-o-up fa-2x" />
      </a>
      <span className="row tag is-medium bold">{props.votes}</span>
      <a 
        className="is-danger is-small row has-text-danger" 
        onClick={props.handleVoteDown}
      >
        <i className="fa fa-thumbs-o-down fa-2x" />
      </a>
    </section>
  </figure>
);

VoteUpOrDown.defaultProps = {
  avatarUrl: '',
  votes: 0,
  handleVoteUp: () => {},
  handleVoteDown: () => {}
};

VoteUpOrDown.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
	handleVoteUp: PropTypes.func.isRequired,
  handleVoteDown: PropTypes.func.isRequired
};

export default VoteUpOrDown;