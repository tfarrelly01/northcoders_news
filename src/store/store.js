import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../reducer/reducer';

const logger = createLogger();

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk, logger),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
      ? window.devToolsExtension()
      : f => f
  )
);

export default store;