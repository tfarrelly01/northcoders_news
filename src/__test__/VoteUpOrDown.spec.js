import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../store/store';

import VoteUpOrDown from '../components/VoteUpOrDown';

describe('VoteUpOrDown Component', () => {
  test('VoteUpOrDown renders correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <VoteUpOrDown />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});