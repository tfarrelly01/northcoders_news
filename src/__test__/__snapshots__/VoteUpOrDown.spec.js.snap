// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteUpOrDown Component VoteUpOrDown renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <VoteUpOrDown
    avatarUrl=""
    handleVoteDown={[Function]}
    handleVoteUp={[Function]}
    votes={0}
  />,
  "nodes": Array [
    <VoteUpOrDown
      avatarUrl=""
      handleVoteDown={[Function]}
      handleVoteUp={[Function]}
      votes={0}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <VoteUpOrDown
          avatarUrl=""
          handleVoteDown={[Function]}
          handleVoteUp={[Function]}
          votes={0}
        />
      </Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <VoteUpOrDown
            avatarUrl=""
            handleVoteDown={[Function]}
            handleVoteUp={[Function]}
            votes={0}
          />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <VoteUpOrDown
          avatarUrl=""
          handleVoteDown={[Function]}
          handleVoteUp={[Function]}
          votes={0}
        />,
        "_debugID": 2,
        "_renderedOutput": <VoteUpOrDown
          avatarUrl=""
          handleVoteDown={[Function]}
          handleVoteUp={[Function]}
          votes={0}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <VoteUpOrDown
      avatarUrl=""
      handleVoteDown={[Function]}
      handleVoteUp={[Function]}
      votes={0}
    />
  </Provider>,
}
`;
